Frameworks/AddressBook/ABRecord.mm

```diff
@@ -80,18 +152,404 @@ ABRecordType ABRecordGetRecordType(ABRecordRef record) {
+                NSLocalizedFailureReasonErrorKey : NSLocalizedString(@"The record should not be null!\n", nil)
+            };
+            *error = (__bridge_retained CFErrorRef)[NSError errorWithDomain:(__bridge NSString*)ABAddressBookErrorDomain
+                                                                       code:kABOperationNotPermittedByStoreError
+                                                                   userInfo:userInfo];
+        }
+
+        return false;
+    }
+
+    _ABContact* person = (__bridge _ABContact*)record;
+
+    if (person.type == kAddressBookReadOnlyContact) {
+        if (error) {
+            NSDictionary* userInfo = @{
+                NSLocalizedDescriptionKey : NSLocalizedString(@"Error setting record value.\n", nil),
+                NSLocalizedFailureReasonErrorKey : NSLocalizedString(@"You can only modify contacts created by your app!\n", nil)
+            };
+            *error = (__bridge_retained CFErrorRef)[NSError errorWithDomain:(__bridge NSString*)ABAddressBookErrorDomain
+                                                                       code:kABOperationNotPermittedByStoreError
+                                                                   userInfo:userInfo];
+        }
+
+        return false;
+    }
+
+    // Cases for various name properties.
+    if (contactProperty == kABPersonFirstNameProperty) {
+        NSString* firstName = (__bridge NSString*)value;
+        if (!_checkLength(firstName, kABPersonFirstNameLength, error, @"first name")) {
+            return false;
+        }
+
+        person.contact.firstName = firstName;
+    } else if (contactProperty == kABPersonLastNameProperty) {
+        NSString* lastName = (__bridge NSString*)value;
+        if (!_checkLength(lastName, kABPersonLastNameLength, error, @"last name")) {
+            return false;
+        }
+
+        person.contact.lastName = lastName;
+    } else if (contactProperty == kABPersonMiddleNameProperty) {
+        NSString* middleName = (__bridge NSString*)value;
+        if (!_checkLength(middleName, kABPersonMiddleNameLength, error, @"middle name")) {
+            return false;
+        }
+
+        person.contact.middleName = middleName;
+    } else if (contactProperty == kABPersonPrefixProperty) {
+        NSString* prefix = (__bridge NSString*)value;
+        if (!_checkLength(prefix, kABPersonPrefixLength, error, @"prefix")) {
+            return false;
+        }
+
+        person.contact.honorificNamePrefix = prefix;
+    } else if (contactProperty == kABPersonSuffixProperty) {
+        NSString* suffix = (__bridge NSString*)value;
+        if (!_checkLength(suffix, kABPersonSuffixLength, error, @"suffix")) {
+            return false;
+        }
+
+        person.contact.honorificNameSuffix = suffix;
+    } else if (contactProperty == kABPersonNicknameProperty) {
+        NSString* nickname = (__bridge NSString*)value;
+        person.contact.nickname = nickname;
```

Comment:
No (at least not a documented one)

---
In reply to: [77101290](https://github.com/Microsoft/WinObjC/pull/935#discussion_r77101290) [](ancestors = 77101290)